### PR TITLE
fix: scroll for outline

### PR DESCRIPTION
**Pull request by luka-nextcloud**
### 📝 Summary

* Resolves: #3585

#### 🖼️ Screenshots

🏚️ Before | 🏡 After
---|---
image | [outline-scroll.webm](https://user-images.githubusercontent.com/89908051/228292210-7386a3fd-5ac7-414b-b56e-a8b555363dbe.webm)

### 🚧 TODO

- [ ] ...

### 🏁 Checklist

- [ ] Code is properly formatted (`npm run lint` / `npm run stylelint` / `composer run cs:check`)
- [ ] [Sign-off message](https://probot.github.io/apps/dco/) is added to all commits
- [ ] [Tests](https://github.com/nextcloud/text#-testing-the-app) (unit, integration and/or end-to-end) passing and the changes are covered with tests
- [ ] Documentation ([README](https://github.com/nextcloud/text/blob/main/README.md) or [documentation](https://github.com/nextcloud/documentation/blob/master/admin_manual/configuration_server/text_configuration.rst#L2)) has been updated or is not required

### CODE REVIEW

File: src/components/Editor/TableOfContents.vue [hunk 1]
```diff
@@ -69,6 +69,9 @@ export default {
 	padding: 0 10px;
 	color: var(--color-main-text-maxcontrast);
 	--animation-duration: 0.8s;
+	max-height: 73vh;
+	overflow-x: hidden;
```
Comment (mejo-) [inline src/components/Editor/TableOfContents.vue:73]: Mh, does this work with long headings? What I meant is that the scrollbar should not be there if it is not needed for viewing the whole heading, but for sure it should be there once the heading is too long for being displayed completely within the container without scrolling.

**Review by mejo- — CHANGES_REQUESTED** [review]

File: src/components/Editor/TableOfContents.vue [hunk 2]
```diff
@@ -69,6 +69,8 @@ export default {
 	padding: 0 10px;
 	color: var(--color-main-text-maxcontrast);
 	--animation-duration: 0.8s;
+	max-height: 73vh;
```
[the remark this answers is not in the document]
Comment (mejo-) [reply src/components/Editor/TableOfContents.vue:72]: I also cannot think of a solution that doesn't involve a fixed (relative) height. Without a fixed height, the container will just grow as much as needed to list all headings. But `73vh` already is relative to the window height, so I think it should be ok. Do you have another idea how to tackle this, @juliushaertl?

[the remark this answers is not in the document]
Comment (juliushaertl) [reply src/components/Editor/TableOfContents.vue:72]: I think I would rather make the `.editor-outline` scrollable (which it should be already it seems) and limit its height to `calc(100% - 90px)`. Note 90px is just a dummy value as it seems that the outline is also positioned a bit wrongly in terms of the top offset on master.

**Comment by luka-nextcloud** [comment]
@juliushaertl @mejo- Please check again.

File: src/components/Editor/EditorOutline.vue [hunk 3]
```diff
@@ -55,9 +55,10 @@ export default {
 <style lang="scss" scoped>
 .editor--outline {
 	width:  300px;
-	padding: 0 10px 10px 10px;
+	padding: 0 10px 100px 10px;
 	position: fixed;
 	overflow: auto;
+	height: calc(100% - 100px);
```
Comment (mejo-) [inline src/components/Editor/EditorOutline.vue:61]: This probably still should be `max-height`, to not let the container grow unnecessarily for very short outlines?

**Comment by mejo-** [comment]
Please note that one problem remains even with my changes: the `max-height` of the outline container is not really dynamic. If Text gets used somewhere with even more vertical offset to the window top, it might not work. But the only other option I see is to set `max-height` dynamically using Javascript by calculating the vertical offset. Maybe @juliushaertl or @max-nextcloud have a better idea?

**Comment by mejo-** [comment]
After discussing this again with @juliushaertl, we agreed to merge the PR as it is for now. At least it fixes the issue for Text in Viewer, standalone and in Collectives.

**Review by mejo- — APPROVED** [review]
@luka-nextcloud I took the liberty to rebase and squash the commits.

**Comment by mejo-** [comment]
/backport b741f9237e4b5b1335fd02d33b333e366703e57e to stable26

**Comment by mejo-** [comment]
/backport b741f9237e4b5b1335fd02d33b333e366703e57e to stable25

**Comment by mejo-** [comment]
/backport /b741f9237e4b5b1335fd02d33b333e366703e57e,e5f57db008e73f40a7b00db5ff57bf2dcc7753cc to stable26

**Pull request merged**